frontend/controller/actions/utils.js

```diff
@@ -357,15 +357,21 @@ export async function syncContractsInOrder (groupedContractIDs: Object): Promise
     const index = contractSyncPriorityList.indexOf(key)
     return index === -1 ? contractSyncPriorityList.length : index
   }
-
   try {
-    // $FlowFixMe[incompatible-call]
-    await Promise.all(Object.entries(groupedContractIDs).sort(([a], [b]) => {
-      // Sync contracts in order based on type
+    const sortedContractTypes = Object.entries(groupedContractIDs).sort(([a], [b]) => {
       return getContractSyncPriority(a) - getContractSyncPriority(b)
-    }).map(([, ids]) => {
-      return sbp('chelonia/contract/sync', ids)
-    }))
+    })
+    for (const [, contractIDs] of sortedContractTypes) {
+      // For each contract of this type, check if it still exists before syncing because
+      // e.g. syncing a group contract could have removed one of the chatroom contracts
+      // $FlowFixMe[incompatible-type]
+      for (const contractID of contractIDs) {
+        const { contracts } = sbp('chelonia/rootState')
+        if (contractID in contracts) {
+          await sbp('chelonia/contract/sync', contractID)
```

Comment:
Apart from 'check if it still exists before syncing because' (which IMO is fine), the code isn't equivalent (this is caused by removing `Promise.all`). With `Promise.all`, an exception will be thrown if there's an error, but processing will otherwise continue. With these changes, processing will halt upon the first error. I think this is a mistake, because we want to sync as many things as we possibly can.

Now, an issue is that `Promise.all` will return *before* all promises complete (if there was an error). Using, e.g., `Promise.allSettled` or a try-catch here would do.

Reply:
Changes pushed!